### PR TITLE
Make alias feature visible even if no domain name

### PR DESCRIPTION
**Pull request by sga-odoo**
Task: https://www.odoo.com/web#id=1864218&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
PAD: https://pad.odoo.com/p/r.6f7d1b0f491b5a4345d12c4ac983a487

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_alias.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ class Alias(models.Model):
              "- everyone: everyone can post\n"
              "- partners: only authenticated partners\n"
              "- followers: only followers of the related document or members of following channels\n")
+    incoming_email = fields.Boolean('Incoming Email', help="If checked, You can enable email alias.")
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_alias.py]: This field has nothing to do on the mail.alias model.

File: addons/mail/models/mail_alias.py [hunk 2]
```diff
@@ -206,13 +207,22 @@ def get_alias_values(self):
...
+            vals['alias_name'] = False
         record = super(AliasMixin, self.with_context(
             alias_model_name=self.get_alias_model_name(vals),
             alias_parent_model_name=self._name,
         )).create(vals)
         record.alias_id.sudo().write(record.get_alias_values())
         return record
 
+    @api.multi
+    def write(self, vals):
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_alias.py]: Please remove that kind of code. Aliases don't have to be managed with magic fields like that.

File: addons/mail/models/mail_alias.py [hunk 3]
```diff
@@ -281,3 +281,26 @@ def _alias_check_contact_on_record(self, record, message, message_dict, alias):
                 'error_message': _('restricted to known authors')
             }
         return True
+
+
+class AliasActivationMixin(models.AbstractModel):
+    """ This mixin allows to make mail aliases active or inactive.
+    """
+    _name = 'mail.alias.activation.mixin'
```
[the remark this answers is not in the document]
Comment (mba-odoo) [reply addons/mail/models/mail_alias.py]: @Xavier-Do , we already discussed this point in task chatter with @tde-banana-odoo , he suggested to keep is separate as it does not belong to the alias definition so keep it separate. for more refer task chatter. Thanks

File: addons/mail/models/mail_alias.py [hunk 4]
```diff
@@ -281,3 +281,26 @@ def _alias_check_contact_on_record(self, record, message, message_dict, alias):
...
+    _description = "Active Email Alias"
+
+    alias_active = fields.Boolean('Incoming Email', help="If checked, You can enable email alias.")
```
[the remark this answers is not in the document]
Comment (mba-odoo) [reply addons/mail/models/mail_alias.py]: @Xavier-Do , Agree with you. 👍

File: addons/account/models/account.py [hunk 5]
```diff
@@ -396,6 +396,7 @@ class AccountJournal(models.Model):
     _name = "account.journal"
     _description = "Journal"
     _order = 'sequence, type, code'
+    _inherit = ['mail.alias.activation.mixin']
```
Comment (Xavier-Do) [inline addons/account/models/account.py]: mail.alias.activation.mixin logic should be added to mail.alias.mixin
We will apply mail.alias.mixin on account.journal. Do you want to do this in this task?

Comment (sga-odoo) [reply addons/account/models/account.py]: @Xavier-Do ok we will move mail.alias.activation.mixin to mail.alias.mixin. For account.journal the alias is used for purchase journal only. So inheriting mail.alias.mixin in account.journal will create useless alias for other journals also. Do you want me to do this change?

Comment (sga-odoo) [reply addons/account/models/account.py]: @Xavier-Do changes has been done. Thanks!

**Comment by rdeodoo** [comment]
This task was cancelled, you can close it if this code is not needed anymore

**Pull request closed without merging**